### PR TITLE
schema_tables: std::move mutation into the mutation vector

### PR DESCRIPTION
**Pull request by bhalevy**
To save a copy.

### CODE REVIEW

File: db/schema_tables.cc [hunk 1]
```diff
@@ -2457,7 +2457,7 @@ std::vector<mutation> make_create_aggregate_mutations(schema_features features,
     m.set_clustered_cell(ckey, "return_type", aggregate->return_type()->as_cql3_type().to_string(), timestamp);
     m.set_clustered_cell(ckey, "state_func", aggregate->sfunc()->name().name, timestamp);
     m.set_clustered_cell(ckey, "state_type", state_type->as_cql3_type().to_string(), timestamp);
-    std::vector<mutation> muts = {m};
+    std::vector<mutation> muts = {std::move(m)};
```
Comment (avikivity) [inline db/schema_tables.cc:2460]: There's a chance this invokes the vector(initializer_list) constructor. You can't move from an initializer_list.

Comment (bhalevy) [reply db/schema_tables.cc:2460]: It does.  godbolt.org tells me that indeed `m` is moved to the initializer_list and then it is copied into the vector.
I'll change that to muts.emplace_back(std::move(m)) to save the copy.

Comment (bhalevy) [reply db/schema_tables.cc]: fixed that (and more) in v2

**Comment by bhalevy** [comment]
In v2 ([5fc39de](https://github.com/scylladb/scylladb/commit/5fc39debd21f3ba4d0019eb1caa6a144ecbc1a04)):
- used `muts.emplace_back(std::move(m))` to save copy of the mutation from initializer_list, plus some more copies that could be saved.

**Comment by bhalevy** [comment]
Enhancement.  No backport required.

**Pull request closed without merging**